ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -262,20 +353,6 @@ func validateRayCluster(admissionReview *admissionv1.AdmissionReview) (*admissio
 	workerGroupSpecs := raycluster.Spec.WorkerGroupSpecs
 	for i := 0; i < len(workerGroupSpecs); i++ {
 		workerGroupSpec := workerGroupSpecs[i]
-		if containerRequestingTPUs(workerGroupSpec.Template.Spec.Containers...) {
-			klog.V(1).InfoS("validateRayCluster", "RayCluster", namespace+"/"+clusterName, "Worker Group", workerGroupSpec.GroupName, "Requests TPUs", true)
-			replicas := int(*workerGroupSpec.Replicas)
-			numOfHosts := workerGroupSpec.NumOfHosts
-			for replicaIndex := 0; replicaIndex < replicas; replicaIndex++ {
-				// reset past sliceToWorkers entries for slice in ray cluster
-				groupName := workerGroupSpec.GroupName
-				podSlice := slice{clusterName, groupName, namespace, replicaIndex, numOfHosts}
-				sliceToWorkers[podSlice] = nil
-			}
-		} else {
-			// RayCluster worker group does not request TPUs
-			klog.V(1).InfoS("validateRayCluster", "RayCluster", namespace+"/"+clusterName, "Worker Group", workerGroupSpec.GroupName, "Requests TPUs", false)
-		}
 		// validate NumOfHosts for worker group matches topology nodeSelector
 		workersMatchTopology, err := checkWorkersMatchTopology(clusterName, namespace, workerGroupSpec)
```

Comment:
Can we break out the "Has TPUs" check outside of `checkWorkersMatchTopology` similar to what we did for `mutatePod`? Maybe a new function `checkWorkersRequestTPUs` that runs before the topology check. 

Because TPU webhook intercepts all RayClusters (not just ones requesting TPUs), we should structure the admission logic such that future changes cannot impact admission for non-TPU pods. Right now a bug in checkWorkersMatchTopology could potentially impact GPU pods

Reply:
Done in [4164563](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/41645633cdec3a0e2eae88f650791a57eb945b48), `checkWorkersMatchTopology` was already doing the same check and returning `true` if no TPUs are requested, but I think it makes sense to break even earlier than that to avoid any potential bugs.